fish-rust/src/fish.rs

```diff
@@ -0,0 +1,877 @@
+    }
+
+    // Clear signals in case we were interrupted (#9024).
+    signal_clear_cancel();
+
+    if !opts.batch_cmds.is_empty() {
+        // Run the commands specified as arguments, if any.
+        if ffi::get_login() {
+            // Do something nasty to support OpenSUSE assuming we're bash. This may modify cmds.
+            fish_xdm_login_hack_hack_hack_hack(&mut opts.batch_cmds, &args[my_optind..]);
+        }
+
+        // Pass additional args as $argv.
+        // Note that we *don't* support setting argv[0]/$0, unlike e.g. bash.
+        // PORTING: the args were converted to WString here in C++
+        let list = &args[my_optind..];
+        parser.get_vars().set(
+            L!("argv"),
+            EnvMode::default(),
+            list.iter().map(|&s| s.to_owned()).collect(),
+        );
+        res = run_command_list(parser, &opts.batch_cmds);
+        parser.libdata_pod().exit_current_script = false;
+    } else if my_optind == args.len() {
+        // Implicitly interactive mode.
+        if opts.no_exec && unsafe { libc::isatty(libc::STDIN_FILENO) != 0 } {
+            FLOGF!(
+                error,
+                "no-execute mode enabled and no script given. Exiting"
+            );
+            // above line should always exit
+            return libc::EXIT_FAILURE;
+        }
+        res = ffi::reader_read_ffi(parser.pin(), c_int(libc::STDIN_FILENO)).into();
+    } else {
+        // C++ had not converted at this point, we must undo
+        let n = wcs2string(args[my_optind]);
+        let path = OsStr::from_bytes(&n);
+        my_optind += 1;
+        // Rust sets cloexec by default, see above
+        // We don't need autoclose_fd_t when we use File, it will be closed on drop.
+        match File::options().read(true).open(path) {
+            Err(e) => {
+                FLOGF!(
+                    error,
+                    wgettext!("Error reading script file '%s':"),
+                    path.to_string_lossy()
+                );
+                eprintln!("{}", e);
+            }
+            Ok(f) => {
+                // PORTING: the args were converted to WString here in C++
+                let list = &args[my_optind..];
+                parser.get_vars().set(
+                    L!("argv"),
+                    EnvMode::default(),
+                    list.iter().map(|&s| s.to_owned()).collect(),
+                );
+                let rel_filename = &args[my_optind - 1];
+                // PORTING: this used to be `scoped_push`
+                let old_filename = parser.pin().current_filename_ffi().from_ffi();
+                parser.pin().set_filename_ffi(rel_filename.to_ffi());
+                res = ffi::reader_read_ffi(parser.pin(), c_int(f.as_raw_fd())).into();
+                if res != 0 {
+                    FLOGF!(
+                        warning,
+                        wgettext_fmt!("Error while reading file %ls\n", path.to_string_lossy())
```

Comment:
do not pass user input as format string to `FLOGF`. Let's use `wgettext!` instead of `wgettext_fmt` and let `FLOGF` do the formatting

Reply:
Yeah, that would have been a huge issue in C. Will fix.

It is very strange to have a literal format-specifier just be passed directly through `gettext`. This should be warned against in the some guidelines, like other weird things https://github.com/fish-shell/fish-shell/pull/9952#discussion_r1293354154, and this is a non-issue with `format!()` since it does not support runtime-specified-format templates